### PR TITLE
new: aws_session_token support

### PR DESCRIPTION
**Pull request by packetzero**
Allows server to send session tokens in TLS config, rather than having osquery having to generate them.

This goes back to this work:
https://github.com/packetzero/osquery_aws_notes

### CODE REVIEW

File: osquery/utils/aws/aws_util.cpp [hunk 1]
```diff
@@ -228,6 +229,20 @@ OsquerySTSAWSCredentialsProvider::GetAWSCredentials() {
...
 
+  // config provides STS creds that includes the token
+  if (FLAGS_aws_session_token.size() > 0) {
+    if (access_key_id_.empty()) {
+      initAwsSdk();
+      access_key_id_ = FLAGS_aws_access_key_id;
+      secret_access_key_ = FLAGS_aws_secret_access_key;
+      session_token_ = FLAGS_aws_session_token;
+      VLOG(1) << "Using provided aws_session_token for id:"
+              << FLAGS_aws_access_key_id;
```
Comment (theopolis) [inline osquery/utils/aws/aws_util.cpp]: Is this `FLAGS_aws_access_key_id` sensitive, such that you would not want it logged?

File: osquery/utils/aws/aws_util.cpp [hunk 2]
```diff
@@ -228,6 +229,20 @@ OsquerySTSAWSCredentialsProvider::GetAWSCredentials() {
   // Grab system time in seconds-since-epoch for token expiration checks.
   size_t current_time = osquery::getUnixTime();
 
+  // config provides STS creds that includes the token
+  if (FLAGS_aws_session_token.size() > 0) {
```
Comment (theopolis) [inline osquery/utils/aws/aws_util.cpp]: Can you write this as `!FLAGS_aws_session_token.empty())`?

File: osquery/utils/aws/aws_util.cpp (hunk 1 above)
Comment (packetzero) [reply osquery/utils/aws/aws_util.cpp]: Good point, it's one of three pieces needed, but no need to log entire token.

File: osquery/utils/aws/aws_util.cpp [hunk 3]
```diff
@@ -228,6 +229,20 @@ OsquerySTSAWSCredentialsProvider::GetAWSCredentials() {
...
   size_t current_time = osquery::getUnixTime();
 
+  // config provides STS creds that includes the token
+  if (!FLAGS_aws_session_token.empty()) {
+    if (access_key_id_.empty()) {
+      initAwsSdk();
+      access_key_id_ = FLAGS_aws_access_key_id;
+      secret_access_key_ = FLAGS_aws_secret_access_key;
+      session_token_ = FLAGS_aws_session_token;
+      VLOG(1) << "Using provided aws_session_token for id starting with :"
```
Comment (theopolis) [inline osquery/utils/aws/aws_util.cpp]: Ah, sorry there's another nitpick here, `with: `, right now there's a space between `with` and `:` and not space between the `:` and the first-8-chars of the key.

**Pull request merged**